prediction_prophet/functions/research.py

```diff
@@ -10,6 +11,9 @@
 from prediction_prophet.functions.search import search
 from pydantic.types import SecretStr
 
+if t.TYPE_CHECKING:
+    from loguru import Logger
```

Comment:
What is this change for? It looks below like it is still always using `logging.Logger` below

Reply:
I added the loguru's Logger below into accepted types, because in PMAT/PMA we use loguru, so we can pass it there without mypy complaining.

```
logger: t.Union[logging.Logger, "Logger"] = logging.getLogger()
```

Reply:
And idk why `logging.Logger | "Logger"` doesn't work in Python, so we need to use the old-school `Union[logging.Logger, "Logger"]`.

Reply:
Ahh, so we can use prophet's `research` in PMA? Makes sense!